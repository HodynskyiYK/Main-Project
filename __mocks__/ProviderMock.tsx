import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}