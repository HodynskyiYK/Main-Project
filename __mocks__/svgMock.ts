// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock