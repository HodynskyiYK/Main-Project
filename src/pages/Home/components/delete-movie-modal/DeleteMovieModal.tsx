import React, { FC } from 'react'
import classnames from 'classnames'
import { Modal } from '../../../../components/modal'
import { Button } from '../../../../components/button'
import { useMovieModalContext } from '../../../../context/movie-modal-context'
import styles from './DeleteMovieModal.module.scss'
import { useAction, useTypedSelector } from '../../../../hooks'

interface IDeleteMovieModal {
    movieId?: number | undefined
}

export const DeleteMovieModal: FC<IDeleteMovieModal> = ({movieId}) => {
    const {deleteMovie, fetchMovies} = useAction()
    const {filterBy, sortBy} = useTypedSelector(state => state.movies)
    const {hideDeleteMovieModal} = useMovieModalContext()

    const confirmButtonHandler = async (id: number | undefined) => {
        if (id) {
            const responseStatus = await deleteMovie(id)
            hideDeleteMovieModal()
            if (+responseStatus <= 299) {
                fetchMovies(filterBy, sortBy)
            }
        }
    }

    return (
        <Modal>
            <div className={styles.deleteMovieCard}>
                <p className={classnames('h1', styles.confirmTitle)}>Delete movie</p>
                <p className={styles.confirmText}>Are you sure you want to delete this movie?</p>
                <p className={styles.confirmButton}>
                    <Button
                        buttonClassName={'confirmBtn'}
                        buttonText={'Confirm'}
                        buttonAction={() => confirmButtonHandler(movieId)}
                    />
                </p>
            </div>
        </Modal>
    )
}
