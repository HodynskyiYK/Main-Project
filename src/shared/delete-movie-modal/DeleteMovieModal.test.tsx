import React from 'react'
import { render } from '@testing-library/react'
import { DeleteMovieModal } from './DeleteMovieModal'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('DeleteMovieModal tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock><DeleteMovieModal/></ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})