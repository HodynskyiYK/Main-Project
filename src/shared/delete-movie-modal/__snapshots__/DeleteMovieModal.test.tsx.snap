// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovieModal tests To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modalContent"
    >
      <button
        class="closeModal"
        data-testid="close-btn"
      />
      <div
        class="deleteMovieCard"
      >
        <p
          class="h1 confirmTitle"
        >
          Delete movie
        </p>
        <p
          class="confirmText"
        >
          Are you sure you want to delete this movie?
        </p>
        <p
          class="confirmButton"
        >
          <button
            class="btn confirmBtn"
          >
            Confirm
          </button>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
